### PR TITLE
refactor(cdk/a11y): Refactor focus indicator theme mixins to match other theme mixins

### PR DESCRIPTION
**Pull request by zelliott**

### CODE REVIEW

File: src/dev-app/theme.scss [hunk 1]
```diff
@@ -69,7 +69,7 @@ $candy-app-theme: mat.define-light-theme((
   @include mdc-slider-theme.color($dark-colors);
 
   // Include the dark theme colors for focus indicators.
-  .demo-strong-focus {
+  &.demo-strong-focus {
```
Comment (zelliott) [inline src/dev-app/theme.scss:72]: This was a bug, AFAICT.

File: src/material/core/focus-indicators/_focus-indicators-theme.scss [hunk 2]
```diff
@@ -5,43 +5,18 @@
...
   }
 }
 
-// Mixin that applies the border color for the focus indicators.
-@mixin _mat-strong-focus-indicators-border-color($color) {
-  .mat-focus-indicator::before {
-    border-color: $color;
+@mixin theme($theme-or-color-config) {
+  $theme: theming.private-legacy-get-theme($theme-or-color-config);
+  @include theming.private-check-duplicate-theme-styles($theme, 'mat-focus-indicators') {
```
Comment (zelliott) [inline src/material/core/focus-indicators/_focus-indicators-theme.scss:16]: Note: The id to `private-check-duplicate-theme-styles` was changed from `mat-strong-focus-indicators` to `mat-focus-indicators` to address a linter error I was getting:

> Missing check for duplicative theme styles.  material/theme-mixin-api
> Please include the duplicate styles check mixin: theming.private-check-duplicate-theme-styles($theme, 'mat-focus-indicators')

**Review by jelbourn — APPROVED** [review]
lgtm

**Comment by wagnermaciel** [comment]
@zelliott Sorry for the delay here. Could you resolve these merge conflicts for us?

**Comment by wagnermaciel** [comment]
@zelliott I changed the `target` label to `minor` because there are merge conflicts between these changes and `11.2.x`. If you'd like these changes to also be included in `11.2.x` and `12.0.x`, please let me know and I will create separate PRs for that. Otherwise, if you're ok with this being `target: minor` we can leave this as is.

**Pull request merged**